### ISSUE TITLE
Is there a way to change how often MAP is calculated?

### ISSUE THREAD

**Issue by mazatov** [AlexeyAB/darknet#6457]
I think by default, MAP is calculated every 2500 iterations. Is there a way to change that? 

As I understand xxx_best.weights is picked as the highest MAP model, which if only calculated every 2500 iterations is not often enough for me. I'd like to change it to every 1000 iterations so I get more fine-tuned value in best.weights.

**Comment by divid3d** [comment]
I think it's related to the training dataset size

**Comment by mazatov** [comment]
Oh, interesting. It was ~ 2500 for me so I thought that was the default. Let me know if you find anything. I still haven't figured a way.

**Comment by ausk** [comment]
In [detector.c](https://github.com/AlexeyAB/darknet/blob/master/src/detector.c#L300-L304)

https://github.com/AlexeyAB/darknet/blob/38a164bcb9e017f8c9c3645a39419320e217545e/src/detector.c#L300-L304

Maybe you can `calc_map_for_each` and `next_map_calc`.

**Comment by mazatov** [comment]
@ausk , nice find! I guess I'd need to recompile it than with the new number... I wish there was a way to adjust it in the config or command line.  Basically to have control over that number `4` there, to control every how many epochs you want to calculate MAP

**Comment by blackCmd** [comment]
> @ausk , nice find! I guess I'd need to recompile it than with the new number... I wish there was a way to adjust it in the config or command line. Basically to have control over that number `4` there, to control every how many epochs you want to calculate MAP

Oh, Plz Author I want that!

**Comment by MKiremitci** [comment]
@mazatov Hi, did you change that calc_map_for_each and next_map_calc and get a result?

**Comment by divid3d** [comment]
Basicly, if you want to calculate mAP every n iterations change `int calc_map_for_each` with your desired n fixed value.
https://github.com/AlexeyAB/darknet/blob/38a164bcb9e017f8c9c3645a39419320e217545e/src/detector.c#L300-L304

**Comment by MKiremitci** [comment]
@divid3d thanks for answer :)  I will change these lines than I will run the "make" command, right? I am newbie :(

**Comment by oubaid2** [comment]
I want to calculate mAP starting from 100 batches, then every 100. What should i change in this code?

`int calc_map_for_each = 4 * train_images_num / (net.batch * net.subdivisions);  // calculate mAP for each 4 Epochs
        calc_map_for_each = fmax(calc_map_for_each, 100);
        int next_map_calc = iter_map + calc_map_for_each;
        next_map_calc = fmax(next_map_calc, net.burn_in);
        //next_map_calc = fmax(next_map_calc, 400);
        if (calc_map) {
            printf("\n (next mAP calculation at %d iterations) ", next_map_calc);
            if (mean_average_precision > 0) printf("\n Last accuracy mAP@0.5 = %2.2f %%, best = %2.2f %% ", mean_average_precision * 100, best_map * 100);
        }`

**Comment by namasang1** [comment]
> I want to calculate mAP starting from 100 batches, then every 100. What should i change in this code?
> 
> `int calc_map_for_each = 4 * train_images_num / (net.batch * net.subdivisions); // calculate mAP for each 4 Epochs calc_map_for_each = fmax(calc_map_for_each, 100); int next_map_calc = iter_map + calc_map_for_each; next_map_calc = fmax(next_map_calc, net.burn_in); //next_map_calc = fmax(next_map_calc, 400); if (calc_map) { printf("\n (next mAP calculation at %d iterations) ", next_map_calc); if (mean_average_precision > 0) printf("\n Last accuracy mAP@0.5 = %2.2f %%, best = %2.2f %% ", mean_average_precision * 100, best_map * 100); }`

Did you solve the problem?
